pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Tools.groovy

```diff
@@ -57,6 +58,23 @@ public class Tools extends MappedClosure<String,Tools> implements Serializable {
     }
 
     /**
+     * Merges the tool entries from another instance into this one, defaulting to the current instance's values.
+     *
+     * @return A list of type/name tuples
+     */
+    public List<List<Object>> mergeToolEntries(@CheckForNull Tools other) {
```

Comment:
Since you annotate the nullability of the parameter you should annotate the return value as well. In this case `@Nonnull`. It confused me in the code below that you before expected something could be null, but now you seemed to know that it was non null.